### PR TITLE
Added Category and Controller, Service and Repository 

### PR DESCRIPTION
**Pull request by holsoni**

### CODE REVIEW

File: src/main/java/com/example/bookstore/controller/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @Operation(summary = "Create new book category",
+            description = "Add new book category to the store")
+
```
Comment (Dimagaa) [inline src/main/java/com/example/bookstore/controller/CategoryController.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/repository/book/BookRepository.java [hunk 2]
```diff
@@ -1,8 +1,12 @@
 package com.example.bookstore.repository.book;
 
 import com.example.bookstore.model.Book;
+import java.util.List;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    List<Book> findAllByCategories_Id(Long categoryId, Pageable pageable);
```
Comment (Dimagaa) [inline src/main/java/com/example/bookstore/repository/book/BookRepository.java]:
```suggestion
    List<Book> findAllByCategoryId(Long categoryId, Pageable pageable);
```

**Review by Dimagaa — APPROVED** [review]
You have excellent code)

File: src/main/java/com/example/bookstore/model/User.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    @Column(nullable = false)
+    private String lastName;
+    private String shippingAddress;
+    @ManyToMany(fetch = FetchType.LAZY)
+    @JoinTable(
+            name = "user_role",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
+    private Set<Role> roles;
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/model/User.java]: It's better to assign implementation of a collection so Hibernate can replace it by proper proxied implementation
```suggestion
    private Set<Role> roles = new HashSet<>();
```

File: src/main/java/com/example/bookstore/service/impl/CategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Override
+    public CategoryDto save(CategoryCreateRequestDto categoryDto) {
+        Category category = categoryMapper.toEntity(categoryDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryCreateRequestDto categoryDto) {
+        getById(id);
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/service/impl/CategoryServiceImpl.java:48]: Why do you need to load entity into the persistence context if you don't use it? Better just `exists` method

**Review by vovasalyha — APPROVED** [review]
Approving, but take a look on few comments. Also, I see changes from the previous PR here, try to avoid it in next PRs

**Pull request merged**